File: dev/src/samples/NetSamplesVB/Samples/Styles/DisplayRepresentations.vb
Imports System.Collections.Generic

Imports Autodesk.AutoCAD.DatabaseServices
Imports Autodesk.AutoCAD.Runtime
Imports Autodesk.Civil.Land.DatabaseServices
Imports Autodesk.Civil.Land.DatabaseServices.Styles

<Assembly: CommandClass(GetType(Autodesk.CivilizedDevelopment.DisplayRepresentations))> 

Namespace Autodesk.CivilizedDevelopment
  Public Class DisplayRepresentations
    Inherits SimpleDrawingCommand
    <CommandMethod("CDS_TurnDisplayRepresentationsOn")> _
    Public Sub CDS_TurnDisplayRepresentationsOn()
      Dim styleId As ObjectId = findSurfaceStyle("SampleStyle")
      If styleId = ObjectId.Null Then
        styleId = _civildoc.Styles.SurfaceStyles.Add("SampleStyle")
      End If

      Using tr As Transaction = startTransaction()
        Dim style As SurfaceStyle =
          TryCast(styleId.GetObject(OpenMode.ForWrite), SurfaceStyle)
        Dim settings As SurfaceDisplayStyleType() =
          New SurfaceDisplayStyleType() _
          {
            SurfaceDisplayStyleType.MajorContour,
            SurfaceDisplayStyleType.MinorContour,
            SurfaceDisplayStyleType.Points,
            SurfaceDisplayStyleType.Triangles
          }

        applySettings(style, settings)

        tr.Commit()
      End Using
    End Sub

    Private Function findSurfaceStyle(name As String) As ObjectId
      Using tr As Transaction = startTransaction()
        For Each styleId As ObjectId In _civildoc.Styles.SurfaceStyles
          Dim style As SurfaceStyle = TryCast(styleId.GetObject(OpenMode.ForRead), SurfaceStyle)
          If style.Name = name Then
            Return styleId
          End If
        Next
      End Using

      Return ObjectId.Null
    End Function

    Private Sub applySettings(style As SurfaceStyle, _
                              settings As IList(Of SurfaceDisplayStyleType))
      Dim displayTypes As IEnumerable(Of SurfaceDisplayStyleType) =
        TryCast([Enum].GetValues(GetType(SurfaceDisplayStyleType)), 
          IEnumerable(Of SurfaceDisplayStyleType))

      For Each displayType As SurfaceDisplayStyleType In displayTypes
        Dim state As Boolean = settings.Contains(displayType)
        style.GetDisplayStylePlan(displayType).Visible = state
        style.GetDisplayStyleModel(displayType).Visible = state
      Next
    End Sub
  End Class
End Namespace

File: dev/src/samples/NetSamples/Samples/Styles/DisplayRepresentations.cs
using System;
using System.Collections.Generic;

using Autodesk.AutoCAD.DatabaseServices;
using Autodesk.AutoCAD.Runtime;
using Autodesk.Civil.Land.DatabaseServices;
using Autodesk.Civil.Land.DatabaseServices.Styles;

[assembly: CommandClass(typeof(Autodesk.CivilizedDevelopment.DisplayRepresentations))]

namespace Autodesk.CivilizedDevelopment
{
    public class DisplayRepresentations : SimpleDrawingCommand
    {
        [CommandMethod("CDS_TurnDisplayRepresentationsOn")]
        public void CDS_TurnDisplayRepresentationsOn()
        {
            ObjectId styleId = findSurfaceStyle("SampleStyle");
            if (styleId == ObjectId.Null)
            {
                styleId = _civildoc.Styles.SurfaceStyles.Add("SampleStyle");
            }

            using (Transaction tr = startTransaction())
            {
                SurfaceStyle style = styleId.GetObject(OpenMode.ForWrite) 
                    as SurfaceStyle;
                SurfaceDisplayStyleType[] settings =
                    new SurfaceDisplayStyleType[]
                    {
                        SurfaceDisplayStyleType.MajorContour,
                        SurfaceDisplayStyleType.MinorContour,
                        SurfaceDisplayStyleType.Points,
                        SurfaceDisplayStyleType.Triangles
                    };

                applySettings(style, settings);

                tr.Commit();
            }
        }

        private ObjectId findSurfaceStyle(string name)
        {
            using (Transaction tr = startTransaction())
            {
                foreach (ObjectId styleId in _civildoc.Styles.SurfaceStyles)
                {
                    SurfaceStyle style = styleId.GetObject(OpenMode.ForRead)
                        as SurfaceStyle;
                    if (style.Name == name)
                    {
                        return styleId;
                    }
                }
            }

            return ObjectId.Null;
        }

        private void applySettings(SurfaceStyle style, 
            IList<SurfaceDisplayStyleType> settings)
        {
            IEnumerable<SurfaceDisplayStyleType> displayTypes =
                Enum.GetValues(typeof(SurfaceDisplayStyleType)) 
                as IEnumerable<SurfaceDisplayStyleType>;

            foreach (SurfaceDisplayStyleType displayType in displayTypes)
            {
                bool state = settings.Contains(displayType);
                style.GetDisplayStylePlan(displayType).Visible = state;
                style.GetDisplayStyleModel(displayType).Visible = state;
            }
        }
    }
}